extensions/wikia/FilePage/WikiaFilePage.php

```diff
@@ -148,4 +148,24 @@ protected function newPage( Title $title ) {
 		return new WikiaWikiFilePage( $title );
 	}
 
+	/**
+	 * Render the image or video
+	 */
+	public function view() {
+		global $wgRedirectFilePagesForAnons;
+		if ( !$this->getContext()->getUser()->isAnon() || empty( $wgRedirectFilePagesForAnons ) ) {
+			parent::view();
+
+			return;
+		}
+		$data = F::app()->sendRequest(
```

Comment:
I don't think this should be using a controller and `sendRequest` to get this redirect URL. The controller would be useful if this was also used externally or was doing something that required Nirvana controllers. For this purpose, I think this would be cleaner if `fileRedir` was in just a helper class (`FilePageHelper` is fine) and you could just call this with `$url = ( new FilePageHelper() )->fileRedir( $fileTitle );`. Rather than `F::app()->sendRequest` in multiple places that don't need it to be a Nirvana controller.